src/backends/torch/torchgraphbackend.h

```diff
@@ -103,27 +103,31 @@ namespace dd
     /**
      * \brief allocates torch modules base on real input dimension
      * @param inputdim input tensor dimension, in int64_t
+     * @return true if some allocation was done
      */
-    void finalize(std::vector<int64_t> inputdim);
+    bool finalize(std::vector<int64_t> inputdim);
```

Comment:
Instead of returning a `bool`, there could be a flag (`need_reload` or something else) and a method `needs_reload()` to query this flag. This would avoid returning an unclear value (that needs to be documented) and make things more explicit in my opinion. And the graph could keep its "dirty" state in memory, so we do not need to keep track of it externally and it's more flexible to use (e.g. we do not need to save the result of the `finalize` method in a variable or a class field if we want to perform loading later in the code).

Reply:
ok done